libr/anal/esil_interrupt.c

```diff
@@ -136,12 +214,60 @@ R_API bool r_anal_esil_load_interrupts_from_lib(RAnalEsil *esil, const char *pat
 	return r_anal_esil_load_interrupts (esil, handlers, src_id);
 }
 
-R_API void r_anal_esil_interrupts_fini(RAnalEsil *esil) {
-	if (esil && esil->interrupts) {
-		_interrupt_free_cb (esil->intr0);
-		esil->intr0 = NULL;
-		esil->interrupts->f = _interrupt_free_cb;
-		dict_free (esil->interrupts);
-		esil->interrupts = NULL;
+R_API bool r_anal_esil_load_syscalls_from_lib(RAnalEsil *esil, const char *path) {
+	r_return_val_if_fail (esil, false);
+	const ut32 src_id = r_anal_esil_load_source (esil, path);
+	if (!src_id) {
+		return false;
+	}
+	RAnalEsilHandler **handlers = (RAnalEsilHandler **)
+		r_lib_dl_sym (r_anal_esil_get_source (esil, src_id), "syscalls");
```

Comment:
imho its a bad idea to use low level r_lib_dl this can lead to resolving symbols on NULL and it would be better if we could define syscalls without depending on native code. maybe having an rlang interface for it would be ideal to use py/js or other safe languages to do so. What do you think=?

Reply:
I want syscall handlers that can interact with libr to full extend. They're supposed to simulate lowlevel things that the OS does